Python bindings fail to build on macOS

```
./dnet.c:781:10: fatal error: 'bsd/string.h' file not found
#include "bsd/string.h"
         ^~~~~~~~~~~~~~
1 error generated.
```
`strlcpy` and `strlcat` are declared in `string.h`. I'm not sure that `bsd/string.h` exists on BSD systems either, it might just be a Linux thing. You probably want to include `string.h` and then also include `bsd/string.h` if it exists.

The branch still fails the same way I'm afraid.
```
./dnet.c:782:10: fatal error: 'bsd/string.h' file not found
#include "bsd/string.h"
         ^~~~~~~~~~~~~~
1 error generated.
```

There's no getting around the need to check whether the header exists before including it. Clang offers the `__has_include` macro, but it was only added to GCC very recently. The most portable way would still be to use `AC_CHECK_HEADER` in configure and substitute the result into either `setup.py` or `dnet.pyx`, and then include the header conditionally based on that. Something along the lines of:
```
#ifdef HAVE_BSD_STRING_H
#include <bsd/string.h>
#endif
```

--- a/configure.ac
+++ b/configure.ac
@@ -190,6 +190,13 @@ else
 	    netinet/ip_compat.h ip_fil.h netinet/ip_fil.h)
 	AC_CHECK_HEADERS(hpsecurity.h stropts.h)
 fi
+AC_CHECK_HEADERS(bsd/string.h)
+if test "$ac_cv_header_bsd_string_h" = yes; then
+    STRLCPY_HEADER="bsd/string.h"
+else
+    STRLCPY_HEADER="string.h"
+fi
+AC_SUBST(STRLCPY_HEADER)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
@@ -336,5 +343,6 @@ fi
 
 AC_OUTPUT([Makefile dnet-config include/Makefile include/dnet/Makefile
 	man/Makefile src/Makefile python/Makefile python/setup.py
-	test/Makefile test/check/Makefile test/dnet/Makefile],
+	python/dnet.pxd test/Makefile test/check/Makefile
+	test/dnet/Makefile],
 	[chmod 755 dnet-config])

--- a/python/dnet.pxd.in
+++ b/python/dnet.pxd.in
@@ -1,0 +1,2 @@
+cdef extern from "@STRLCPY_HEADER@":
+    size_t     strlcpy(char *dst, char *src, size_t size)

--- a/python/dnet.pyx
+++ b/python/dnet.pyx
@@ -40,8 +40,7 @@ cdef extern from *:
     unsigned long htonl(unsigned long n)
     unsigned long ntohl(unsigned long n)
 
-cdef extern from "bsd/string.h":
-    int     strlcpy(char *dst, char *src, int size)
+from dnet cimport *
 
 cdef __memcpy(char *dst, object src, int n):
     if PyBytes_Size(src) != n:

